dht_test.go

```diff
@@ -1519,6 +1563,11 @@ func TestFixLowPeers(t *testing.T) {
 		mainD.routingTable.RemovePeer(d.self)
 	}
 
+	// remove blacklist of already contacted peers
+	mainD.recentlyCheckedPeersLk.Lock()
+	mainD.recentlyCheckedPeers = make(map[peer.ID]time.Time)
```

Comment:
https://pkg.go.dev/golang.org/x/exp/maps#Clear I know this is "experimental" but this is stable now, and most likely shipping in 1.21 (it's in the master branch).

Clear conveys better the meaning of what you intended here than overwritting and relying on the GC to clear after you.